pyquil/quilatom.py

```diff
@@ -875,49 +899,40 @@ class Frame(quil_rs.FrameIdentifier):
     Representation of a frame descriptor.
     """
 
-    @staticmethod
-    def __new__(cls, qubits: Sequence[Union[int, Qubit, FormalArgument]], name: str) -> "Frame":
+    def __new__(cls, qubits: Sequence[QubitDesignator], name: str) -> Self:
         return super().__new__(cls, name, _convert_to_rs_qubits(qubits))
 
     @classmethod
     def _from_rs_frame_identifier(cls, frame: quil_rs.FrameIdentifier) -> "Frame":
-        return cls(frame.qubits, frame.name)
+        return super().__new__(cls, frame.name, frame.qubits)
 
-    @property
-    def qubits(self) -> Tuple[QubitDesignator]:
+    @property  # type: ignore[override]
+    def qubits(self) -> Tuple[QubitDesignator, ...]:
         return tuple(_convert_to_py_qubits(super().qubits))
 
     @qubits.setter
-    def qubits(self, qubits: Tuple[Qubit, FormalArgument]):
-        return quil_rs.FrameIdentifier.qubits.__set__(self, _convert_to_rs_qubits(qubits))
-
-    @property
-    def name(self) -> str:
-        return super().name
-
-    @name.setter
-    def name(self, name: str):
-        return quil_rs.FrameIdentifier.name.__set__(self, name)
+    def qubits(self, qubits: Tuple[Qubit, FormalArgument]) -> None:
+        quil_rs.FrameIdentifier.qubits.__set__(self, _convert_to_rs_qubits(qubits))  # type: ignore[attr-defined]
```

Comment:
For whatever reason, mypy doesn't pick up on the `__set__` method being defined. If I understand correctly, it _should_ be implied by the `@property_name.setter` decorator on the base class. I believe it may be related to [this mypy issue](https://github.com/python/mypy/issues/1465), but it's hard to be sure. Either way, if it gets fixed by mypy at some point, it will start throwing errors at us for unused `# type: ignore` comments.